eng/scripts/Automation-Sdk-UpdateChangelog.ps1

```diff
@@ -0,0 +1,616 @@
+    $pinfo.RedirectStandardError = $true
+    $pinfo.RedirectStandardOutput = $true
+    $pinfo.UseShellExecute = $false
+    $pinfo.Arguments = $mvnArgs -join " "
+    
+    $process = New-Object System.Diagnostics.Process
+    $process.StartInfo = $pinfo
+    $process.Start() | Out-Null
+    
+    $stdout = $process.StandardOutput.ReadToEnd()
+    $stderr = $process.StandardError.ReadToEnd()
+    $process.WaitForExit()
+    $exitCode = $process.ExitCode
+    
+    if ($exitCode -ne 0) {
+        LogError "Changelog generation failed with exit code $exitCode"
+        LogError "Standard Output:"
+        LogError $stdout
+        LogError "Error Output:"
+        LogError $stderr
+        throw "Changelog generation failed"
+    }
+    
+    # Parse the JSON output
+    try {
+        $changelogJson = $stdout | ConvertFrom-Json
+        return $changelogJson
+    }
+    catch {
+        LogError "Failed to parse changelog JSON output"
+        LogError "Output: $stdout"
+        throw
+    }
+}
+
+# Function to generate structured changelog content from raw changelog text
+function New-ChangelogContent {
+    <#
+    .SYNOPSIS
+        Parses raw changelog text into structured content with sections.
+    
+    .DESCRIPTION
+        Takes raw changelog text and parses it into structured arrays containing
+        ReleaseContent (all lines) and Sections (organized by section headers).
+        This function only generates content structure without modifying any files.
+    
+    .PARAMETER NewChangelogText
+        The new changelog text containing sections (e.g., "### Breaking Changes", "### Features Added").
+    
+    .PARAMETER InitialAtxHeader
+        The markdown header level used in the changelog (e.g., "#" for H1, "##" for H2).
+    
+    .OUTPUTS
+        PSCustomObject with ReleaseContent and Sections properties.
+    
+    .EXAMPLE
+        $content = New-ChangelogContent -NewChangelogText $changelogText -InitialAtxHeader "#"
+    #>
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$NewChangelogText,
+        
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$InitialAtxHeader
+    )
+    
+    Write-Verbose "Parsing changelog text into structured content..."
+    
+    # Parse the new changelog content into lines
+    $newChangelogLines = $NewChangelogText -split "`r?`n"
+    
+    # Initialize content structure
+    $releaseContent = @()
+    $sections = @{}
+    
+    # Add an empty line after the version header
+    $releaseContent += ""
+    
+    # Parse the new changelog content
+    # InitialAtxHeader represents the markdown header level (e.g., "#" for H1, "##" for H2)
+    # Section headers are one level deeper (e.g., "##" if InitialAtxHeader is "#")
+    $currentSection = $null
+    $sectionHeaderRegex = "^$($InitialAtxHeader)##\s+(?<sectionName>.*)"
+    
+    foreach ($line in $newChangelogLines) {
```

Comment:
This looks like another place it is probably worth creating a helper in the common script to handle this. We can share it with the implementation that is already there in the parser as well.

Reply:
Agree! refer to this comment: https://github.com/Azure/azure-sdk-for-java/pull/47079#discussion_r2540159249